lib/pos_blockchain.go

```diff
@@ -762,15 +763,11 @@ func (bc *Blockchain) validateAndIndexBlockPoS(block *MsgDeSoBlock, parentUtxoVi
 		return bc.storeValidateFailedBlockWithWrappedError(block, err)
 	}
 
-	// Connect this block to the parent block's UtxoView.
-	txHashes := collections.Transform(block.Txns, func(txn *MsgDeSoTxn) *BlockHash {
-		return txn.Hash()
-	})
-
 	// If we fail to connect the block, then it means the block is invalid. We should store it as ValidateFailed.
-	if _, err = parentUtxoView.ConnectBlock(block, txHashes, verifySignatures, nil, block.Header.Height); err != nil {
-		// If it doesn't connect, we want to mark it as ValidateFailed.
-		return bc.storeValidateFailedBlockWithWrappedError(block, err)
+	// TODO: check the error message for specific error for failure to connect block.
+	if _, err = bc.getUtxoViewAtBlockHashWithTipBlock(*blockHash, block); err != nil {
+		return bc.storeValidateFailedBlockWithWrappedError(block, errors.Wrap(
+			err, "validateAndIndexBlockPoS: Problem getting UtxoView"))
```

Comment:
The utxoView cache is nice in its current form because only blocks that are fully validated and stored in badger will ever cached. With this change, if `getUtxoViewAtBlockHashWithTipBlock` succeeds here but fails validation below, then we'll end up with a block in the blockView cache that isn't validated or stored fully in badger.

If an external caller ever wants to construct a view for such a block, the view will be returned, possible giving the impression that the block is fully valid. I'm not sure we ever want to deal with the possibility of that edge case. IMO, it's best to revert this line and leave the caching simple.